### ISSUE TITLE
Redundant information in argparse help descriptions (between "Example:" and default value)

### ISSUE THREAD

**Issue by joshuacwnewton** [spinalcordtoolbox/spinalcordtoolbox#2892]
Often, we manually add `Example: <value>` or `(e.g. <value>)` text in our argparse help descriptions. Sometimes this is just a repeat of the default value, for example:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/120ce12612333efd69c50bb00d27c815e981965d/scripts/sct_analyze_lesion.py#L82-L88

Because default values are now automatically shown, thanks to #2886, the following output is produced:

```
  -ofolder <folder>  Output folder (e.g. "./") (default: ./)
```

I imagine this is happening in other places, too, but I haven't done a thorough investigation yet.

We could just manually edit each description to remove redundant text, `(e.g. "./")`. Although, there might also be a better way to display "Example" vales for argparse rather than manually appending them to the description.

**Comment by Drulex** [comment]
I agree. There is the [`epilog`](https://docs.python.org/3/library/argparse.html#epilog) attribute that can be used to display arbitrary text after the description of arguments. We could aggregate examples in that field?

**Comment by jcohenadad** [comment]
> I agree. There is the epilog attribute that can be used to display arbitrary text after the description of arguments. We could aggregate examples in that field?

epilog is really useful indeed. We have started using it, see eg `sct_qc`

**Comment by cfhammill** [comment]
So is the idea we'd refactor all instances of `Example: ` so that they appear in the epilogue and while we do that refactor remove cases where the example is the same as the default? I see these as two somewhat separate issues, although it might be a good opportunity to make two fixes at once.

I don't immediately see why using `epilog` would be preferable to putting examples in the help, although maybe it makes it more amenable to programmatic manipulation.

**Comment by Drulex** [comment]
> So is the idea we'd refactor all instances of `Example: ` so that they appear in the epilogue and while we do that refactor remove cases where the example is the same as the default?

I think we would all agree on removing duplicate examples as a primary objective.

> I don't immediately see why using epilog would be preferable to putting examples in the help, although maybe it makes it more amenable to programmatic manipulation.

Personally I think that having a dedicated section for examples would encourage us to add more comprehensive examples in that section as opposed to the current (sometimes not so helpful) scoped examples that appear in the `help` section of some arguments. This adds value for users.

**Comment by Drulex** [comment]
For example: https://spinalcordtoolbox.com/en/latest/user_section/command-line.html#sct-image

Currently the examples listed are trivial examples on how to pass file names and list values which aren't really needed since the argument parser is already descriptive. I understand I am not a target user, but I feel that examples showing how to perform common use cases or combine options would be welcomed by users.

**Comment by joshuacwnewton** [comment]
> Personally I think that having a dedicated section for examples would encourage us to add more comprehensive examples in that section as opposed to the current (sometimes not so helpful) scoped examples that appear in the help section of some arguments. This adds value for users.

I initially had similar thoughts as @cfhammill, but I understand better now that you've mentioned "more comprehensive examples".

#2857 is a related idea, just in case the argparse help descriptions start to become too long and unruly.

> Currently the examples listed are trivial examples on how to pass file names and list values which aren't really needed since the argument parser is already descriptive.

Part of why we do this is due to our non-standard formatting for some argument values, too. (We sometimes ask users to delimit their multi-value commands in potentially unintuitive ways that need to be demonstrated.)
